### ISSUE TITLE
Crashlytics fails to propagate the exception to Android's uncaught exception handler

### ISSUE THREAD

**Issue by sebouh00** [firebase/firebase-android-sdk#2869]
### [READ] Step 1: Are you in the right place?

Issues filed here should be about bugs in __the code in this repository__. 
If you have a general question, need help debugging, or fall into some
other category use one of these other channels:

  * For general technical questions, post a question on [StackOverflow](http://stackoverflow.com/)
    with the firebase tag.
  * For general Firebase discussion, use the [firebase-talk](https://groups.google.com/forum/#!forum/firebase-talk)
    google group.
  * For help troubleshooting your application that does not fall under one
    of the above categories, reach out to the personalized
    [Firebase support channel](https://firebase.google.com/support/).

### [REQUIRED] Step 2: Describe your environment

  * Android Studio version: 2020.3.1
  * Firebase Component: Crashlytics
  * Component version: 17.4.1

#### Steps to reproduce:

1. The app is set up to use a package name **for which crashlytics isn't enabled** on Firebase.
2. Logging is enabled using `adb shell setprop log.tag.FirebaseCrashlytics VERBOSE`
3. An uncaught exception is thrown on a background thread, and it propagates to the Crashlytics handler.
4. The following gets logged into the system log:
```
FirebaseCrashlytics  D  Handling uncaught exception "com.example.app.threading.executors.ExecutorException: Uncaught executable exception" from thread pool-8

FirebaseCrashlytics  E  Settings request failed.
                     E  java.io.FileNotFoundException: https://firebase-settings.crashlytics.com/spi/v2/platforms/android/gmp/1:162149437481:android:4329f26bd08ffbd79fcc38/settings?instance=0ec843aa90f82068fe0535c07a55cace7b04fe0b&build_version=27&display_version=1.26.0&source=4
                     E      at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getInputStream(HttpURLConnectionImpl.java:255)
                     E      at com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.getInputStream(DelegatingHttpsURLConnection.java:211)
                     E      at com.android.okhttp.internal.huc.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:30)
                     E      at com.google.firebase.crashlytics.internal.network.HttpGetRequest.execute(HttpGetRequest.java:80)
                     E      at com.google.firebase.crashlytics.internal.settings.network.DefaultSettingsSpiCall.invoke(DefaultSettingsSpiCall.java:113)
                     E      at com.google.firebase.crashlytics.internal.settings.SettingsController$1.then(SettingsController.java:199)
                     E      at com.google.firebase.crashlytics.internal.settings.SettingsController$1.then(SettingsController.java:192)
                     E      at com.google.android.gms.tasks.zzo.run(com.google.android.gms:play-services-tasks@@17.2.0:2)
                     E      at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
                     E      at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
                     E      at com.google.firebase.crashlytics.internal.common.ExecutorUtils$1$1.onRun(ExecutorUtils.java:64)
                     E      at com.google.firebase.crashlytics.internal.common.BackgroundPriorityRunnable.run(BackgroundPriorityRunnable.java:27)
                     E      at java.lang.Thread.run(Thread.java:920)
```
The app stays alive in the background and does not terminate. [The following](https://github.com/firebase/firebase-android-sdk/blob/master/firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/CrashlyticsUncaughtExceptionHandler.java#L59) line does NOT get logged:
```
FirebaseCrashlytics  D  Completed exception processing. Invoking default exception handler.
```
 
It seems to me that the wait [here](https://github.com/firebase/firebase-android-sdk/blob/master/firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/CrashlyticsController.java#L232) goes on forever because the task isn't properly ended.

**Comment by aguatno** [comment]
Hi @sebouh00 thank for reporting. I noticed this error "Settings request failed." usually happens when the app doesn't have Crashlytics enabled in the Firebase dashboard. If you were trying to communicate with Firebase make sure you are using the correct google-services.json file.

**Comment by sebouh00** [comment]
@aguatno Yes, that's clear. It would be great though if it doesn't make the thread wait forever, and if it outputs a logs message saying that Crashlytics isn't enabled.

**Comment by mrichards** [comment]
I reproduced this using `17.4.1`, and have verified that it is no longer a bug in the latest version (currently `18.2.13`). It was likely fixed in [`18.2.10`](https://firebase.google.com/support/release-notes/android#crashlytics_v18-2-10).
